lib/phoenix_live_view/html_tokenizer.ex

```diff
@@ -103,8 +103,8 @@ defmodule Phoenix.LiveView.HTMLTokenizer do
     handle_text(rest, line, column + 1, [char_or_bin(c) | buffer], acc, state)
   end
 
-  defp handle_text(<<>>, line, column, buffer, acc, _state) do
-    ok(text_to_acc(buffer, acc, line, column, []), :text)
+  defp handle_text(<<>>, line, column, buffer, acc, state) do
+    ok(text_to_acc(buffer, acc, line, column, state.context), :text)
```

Comment:
The fix actually is in the tokenizer making sure it will sent the `:comment_end` for this case.